#F2 HC RNA-Seq Dataset
#01_Reading In MetaData
#Megan Hagenauer and Elaine Hebda-Bauer
#2022-01-04, updated later for a few figures for the paper.


#set working directory
setwd("\\\\n05-corea-cifs.umhs.med.umich.edu/Home1/Users/hebda/My Documents/F2_TMMnorm_20220104updated")
getwd()

#Package and version information (updated 1/21/2022):
sessionInfo()  
# R version 4.1.1 (2021-08-10)
# Platform: i386-w64-mingw32/i386 (32-bit)
# Running under: Windows 10 x64 (build 19042)
# 
# Matrix products: default
# 
# locale:
#   [1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                           LC_TIME=English_United States.1252    
# 
# attached base packages:
#   [1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
#   [1] fgsea_1.18.0         plyr_1.8.6           org.Rn.eg.db_3.13.0  stringr_1.4.0        car_3.0-12           carData_3.0-5        edgeR_3.34.1         AnnotationDbi_1.54.1 IRanges_2.26.0       S4Vectors_0.30.2     Biobase_2.52.0      
# [12] BiocGenerics_0.38.0  limma_3.48.3        
# 
# loaded via a namespace (and not attached):
#   [1] Rcpp_1.0.7             locfit_1.5-9.4         lattice_0.20-44        png_0.1-7              Biostrings_2.60.2      assertthat_0.2.1       utf8_1.2.2             R6_2.5.1               GenomeInfoDb_1.28.4    RSQLite_2.2.8         
# [11] httr_1.4.2             ggplot2_3.3.5          pillar_1.6.4           zlibbioc_1.38.0        rlang_0.4.11           rstudioapi_0.13        data.table_1.14.2      blob_1.2.2             Matrix_1.4-0           BiocParallel_1.26.2   
# [21] RCurl_1.98-1.5         bit_4.0.4              munsell_0.5.0          compiler_4.1.1         pkgconfig_2.0.3        tidyselect_1.1.1       KEGGREST_1.32.0        gridExtra_2.3          tibble_3.1.6           GenomeInfoDbData_1.2.6
# [31] fansi_1.0.2            crayon_1.4.2           dplyr_1.0.7            bitops_1.0-7           grid_4.1.1             gtable_0.3.0           lifecycle_1.0.1        DBI_1.1.2              magrittr_2.0.1         scales_1.1.1          
# [41] stringi_1.7.5          cachem_1.0.6           XVector_0.32.0         ellipsis_0.3.2         vctrs_0.3.8            generics_0.1.1         fastmatch_1.1-3        tools_4.1.1            bit64_4.0.5            glue_1.6.0            
# [51] purrr_0.3.4            abind_1.4-5            fastmap_1.1.0          colorspace_2.0-2       BiocManager_1.30.16    memoise_2.0.1         

####################

#Reading in the F2 metadata:
#Includes the PCA output from Ahub (performed on data that hadn't been QCed yet)

F2_PCA_wMetaData<-read.delim("HRLR_F2_pca_matrix_RNAconc.txt", sep='\t', header=TRUE, stringsAsFactors = FALSE)

str(F2_PCA_wMetaData)
# 'data.frame':	250 obs. of  50 variables:
# $ row_name                 : chr  "sl470891" "sl470889" "sl470888" "sl470890" ...
# $ PC1                      : num  0.0172 -0.0103 -0.0327 -0.0306 -0.0779 ...
# $ PC2                      : num  -0.0216 0.0156 -0.0454 -0.0599 -0.0193 ...
# $ PC3                      : num  -0.00128 -0.04807 0.04414 0.03775 0.05776 ...
# $ PC4                      : num  0.03889 -0.08052 -0.07268 -0.05395 0.00364 ...
# $ PC5                      : num  0.0915 0.0469 0.0153 0.0106 0.0444 ...
# $ PC6                      : num  0.043 0.0552 -0.0259 0.0442 -0.059 ...
# $ RNAconc                  : num  100 171 146 127 157 ...
# $ PF.Reads                 : int  32337825 30807308 32349335 30101447 26047278 28642846 31525152 31582801 29433809 25294045 ...
# $ RibosomePerc             : num  0.0111 0.0103 0.009 0.0085 0.0072 0.0124 0.0221 0.0139 0.0085 0.0136 ...
# $ mRNAperc                 : num  0.827 0.813 0.81 0.811 0.806 ...
# $ RIN                      : num  8.9 8.9 8.9 8.9 9 8.9 9.5 8.7 9 8.7 ...
# $ DV200                    : num  96.7 96.7 96.8 96.9 96.8 96.7 97 96.8 96.9 96.8 ...
# $ HPC_Dissection_Date      : chr  "12/14/2020" "12/14/2020" "12/14/2020" "12/14/2020" ...
# $ HPC_Dissection_Day       : int  14 14 14 14 1 7 11 10 10 10 ...
# $ HPC_Side_Used            : chr  "L" "R" "L" "R" ...
# $ Dissector                : chr  "EHB" "EHB" "EHB" "EHB" ...
# $ Age.days.                : int  118 118 127 127 132 132 118 118 118 118 ...
# $ Family                   : int  1 1 1 1 1 1 1 1 1 1 ...
# $ Sex                      : chr  "Male" "Male" "Female" "Female" ...
# $ Rat                      : int  324 325 328 329 330 331 344 345 346 347 ...
# $ Lateral_LocoScore        : int  94 209 148 232 382 124 36 146 139 54 ...
# $ Rearing_LocoScore        : int  181 294 228 401 713 242 29 209 194 122 ...
# $ Total_LocoScore          : int  275 503 376 633 1095 366 65 355 333 176 ...
# $ EPM_Percent_Time_Open_Arm: num  0.82 8.18 24.19 33.08 7.19 ...
# $ EPM_Boli                 : int  4 0 0 0 0 0 0 2 1 5 ...
# $ EPM_DistanceTraveled     : num  1428 1851 2574 2471 1807 ...
# $ EPM_Time_Immobile        : num  167 120.7 71.7 81.4 99.4 ...
# $ LearningClassification   : chr  "" "" "" "" ...
# $ PCA_Index_Days6and.7     : num  NA NA NA NA 0.863 ...
# $ Response_Bias_Day7       : num  NA NA NA NA 0.939 ...
# $ Prob_Diff_Day7           : num  NA NA NA NA 0.92 0.64 -0.8 -0.52 -0.8 -0.12 ...
# $ Latency_Score_Day7       : num  NA NA NA NA 0.614 ...
# $ PCA_Index_Day7only       : num  NA NA NA NA 0.824 ...
# $ Response_Bias_Day6       : num  NA NA NA NA 1 ...
# $ Prob_Diff_Day6           : num  NA NA NA NA 1 0.32 -0.76 -0.24 -0.8 -0.72 ...
# $ Latency_Score_Day6       : num  NA NA NA NA 0.706 ...
# $ PCA_Index_Day6only       : num  NA NA NA NA 0.902 ...
# $ EPM_RawDataFile          : int  2 2 1 1 2 2 1 1 1 2 ...
# $ EPM_Trial                : int  16 17 17 18 32 33 4 5 6 13 ...
# $ EPM_TestDate             : chr  "10/30/2013" "10/30/2013" "10/18/2013" "10/18/2013" ...
# $ EPM_TestDay              : int  7 7 1 1 7 7 1 1 1 7 ...
# $ EPM_TestOrderByDay       : int  16 17 17 18 32 33 4 5 6 13 ...
# $ Loco_TestDate            : chr  "9/24/2013" "9/24/2013" "9/26/2013" "9/26/2013" ...
# $ Loco_TestDay             : int  1 1 3 3 3 3 2 2 2 2 ...
# $ Loco_Batch               : int  1 1 2 2 2 2 1 1 1 1 ...
# $ Loco_Rack                : int  1 1 1 1 1 1 1 1 1 1 ...
# $ Loco_Box                 : int  4 5 2 3 4 5 4 5 6 7 ...
# $ DOB                      : chr  "8/6/2013" "8/6/2013" "8/6/2013" "8/6/2013" ...
# $ DOD                      : chr  "12/4/2013" "12/4/2013" "12/13/2013" "12/13/2013" ...


#Family and Rat number need to be made into characters/factors
#Set reference level for Sex

F2_PCA_wMetaData$Sex_AsFactor<-as.factor(F2_PCA_wMetaData$Sex)
levels(F2_PCA_wMetaData$Sex_AsFactor)
#[1] "Female" "Male"

#Setting the reference levels for factors:
F2_PCA_wMetaData$Sex_AsFactor<-relevel(F2_PCA_wMetaData$Sex_AsFactor, ref="Male")
levels(F2_PCA_wMetaData$Sex_AsFactor)
#[1] "Male"   "Female"

#For including in the covariation matrix easily:
F2_PCA_wMetaData$Sex_AsNumeric<-as.numeric(F2_PCA_wMetaData$Sex_AsFactor)
str(F2_PCA_wMetaData)
# 'data.frame':	250 obs. of  19 variables:
# $ row_name           : chr  "sl469967" "sl469968" "sl469969" "sl469970" ...
# $ PC1                : num  -0.0841 -0.0758 -0.0948 -0.0801 -0.0805 ...
# $ PC2                : num  -0.0449 -0.0408 -0.0114 -0.0233 -0.0261 ...
# $ PC3                : num  0.0896 0.0954 0.1091 0.0727 0.0797 ...
# $ PC4                : num  -0.01801 0.00538 -0.00936 0.02872 0.005 ...
# $ PC5                : num  0.0335 0.1008 -0.0449 -0.0554 0.0442 ...
# $ PC6                : num  -0.1559 -0.1208 -0.1954 0.0721 -0.1129 ...
# $ RNAconc            : num  127.2 98.5 143.9 191.4 125.3 ...
# $ PF.Reads           : int  30800397 27366626 29238544 38111688 30064848 26047278 26871577 27028205 25643674 28446948 ...
# $ RibosomePerc       : num  0.0096 0.0101 0.0086 0.0082 0.0088 0.0072 0.0084 0.0112 0.0075 0.0072 ...
# $ mRNAperc           : num  0.802 0.8 0.782 0.792 0.796 ...
# $ HPC_Dissection_Date: chr  "12/1/2020" "12/1/2020" "12/1/2020" "12/1/2020" ...
# $ HPC_Side_Used      : chr  "L" "R" "L" "R" ...
# $ Dissector          : chr  "KA" "KA" "KA" "KA" ...
# $ Age.days.          : int  121 123 123 123 122 132 130 130 131 130 ...
# $ Family             : int  4 9 9 9 4 1 9 11 3 11 ...
# $ Sex                : chr  "Male" "Male" "Male" "Male" ...
# $ Sex_AsFactor       : Factor w/ 2 levels "Male","Female": 1 1 1 1 1 2 2 2 2 2 ...
# $ Sex_AsNumeric      : num  1 1 1 1 1 2 2 2 2 2 ...


F2_PCA_wMetaData$Dissector_AsFactor<-as.factor(F2_PCA_wMetaData$Dissector)
levels(F2_PCA_wMetaData$Dissector_AsFactor)
#[1] "EHB" "KA"

#Setting the reference levels for factors:
F2_PCA_wMetaData$Dissector_AsFactor<-relevel(F2_PCA_wMetaData$Dissector_AsFactor, ref="EHB")
levels(F2_PCA_wMetaData$Dissector_AsFactor)
#[1] "EHB" "KA"


F2_PCA_wMetaData$Dissector_AsNumeric<-as.numeric(F2_PCA_wMetaData$Dissector_AsFactor)


F2_PCA_wMetaData$HPC_Side_AsFactor<-as.factor(F2_PCA_wMetaData$HPC_Side)
levels(F2_PCA_wMetaData$HPC_Side_AsFactor)
#[1] "L" "R"

F2_PCA_wMetaData$HPC_Side_AsNumeric<-as.numeric(F2_PCA_wMetaData$HPC_Side_AsFactor)


F2_PCA_wMetaData$HPC_Dissection_Date_AsFactor<-as.factor(F2_PCA_wMetaData$HPC_Dissection_Date)
levels(F2_PCA_wMetaData$HPC_Dissection_Date_AsFactor)
#[1] "12/1/2020"  "12/10/2020" "12/11/2020" "12/14/2020" "12/2/2020"  "12/3/2020"  "12/4/2020"  "12/7/2020"  "12/8/2020"  "12/9/2020"

#To keep it from mistaking family and rat ids as meaningful numbers:
F2_PCA_wMetaData$Family<-as.character(F2_PCA_wMetaData$Family)

F2_PCA_wMetaData$Rat<-as.character(F2_PCA_wMetaData$Rat)


levels(F2_PCA_wMetaData$LearningClassification)
F2_PCA_wMetaData$LearningClassification_AsFactor<-as.factor(F2_PCA_wMetaData$LearningClassification)
levels(F2_PCA_wMetaData$LearningClassification_AsFactor)
#[1] ""   "GT" "IN" "ST"
F2_PCA_wMetaData$LearningClassification_AsFactor<-relevel(F2_PCA_wMetaData$LearningClassification_AsFactor, ref="ST")

F2_PCA_wMetaData$STGT_experience<-as.factor((F2_PCA_wMetaData$LearningClassification=="")==FALSE)

F2_PCA_wMetaData$STGT_experience_AsNumeric<-as.numeric(F2_PCA_wMetaData$STGT_experience)


str(F2_PCA_wMetaData)
#'data.frame':	250 obs. of  60 variables:
# $ row_name                       : chr  "sl470891" "sl470889" "sl470888" "sl470890" ...
# $ PC1                            : num  0.0172 -0.0103 -0.0327 -0.0306 -0.0779 ...
# $ PC2                            : num  -0.0216 0.0156 -0.0454 -0.0599 -0.0193 ...
# $ PC3                            : num  -0.00128 -0.04807 0.04414 0.03775 0.05776 ...
# $ PC4                            : num  0.03889 -0.08052 -0.07268 -0.05395 0.00364 ...
# $ PC5                            : num  0.0915 0.0469 0.0153 0.0106 0.0444 ...
# $ PC6                            : num  0.043 0.0552 -0.0259 0.0442 -0.059 ...
# $ RNAconc                        : num  100 171 146 127 157 ...
# $ PF.Reads                       : int  32337825 30807308 32349335 30101447 26047278 28642846 31525152 31582801 29433809 25294045 ...
# $ RibosomePerc                   : num  0.0111 0.0103 0.009 0.0085 0.0072 0.0124 0.0221 0.0139 0.0085 0.0136 ...
# $ mRNAperc                       : num  0.827 0.813 0.81 0.811 0.806 ...
# $ RIN                            : num  8.9 8.9 8.9 8.9 9 8.9 9.5 8.7 9 8.7 ...
# $ DV200                          : num  96.7 96.7 96.8 96.9 96.8 96.7 97 96.8 96.9 96.8 ...
# $ HPC_Dissection_Date            : chr  "12/14/2020" "12/14/2020" "12/14/2020" "12/14/2020" ...
# $ HPC_Dissection_Day             : int  14 14 14 14 1 7 11 10 10 10 ...
# $ HPC_Side_Used                  : chr  "L" "R" "L" "R" ...
# $ Dissector                      : chr  "EHB" "EHB" "EHB" "EHB" ...
# $ Age.days.                      : int  118 118 127 127 132 132 118 118 118 118 ...
# $ Family                         : chr  "1" "1" "1" "1" ...
# $ Sex                            : chr  "Male" "Male" "Female" "Female" ...
# $ Rat                            : chr  "324" "325" "328" "329" ...
# $ Lateral_LocoScore              : int  94 209 148 232 382 124 36 146 139 54 ...
# $ Rearing_LocoScore              : int  181 294 228 401 713 242 29 209 194 122 ...
# $ Total_LocoScore                : int  275 503 376 633 1095 366 65 355 333 176 ...
# $ EPM_Percent_Time_Open_Arm      : num  0.82 8.18 24.19 33.08 7.19 ...
# $ EPM_Boli                       : int  4 0 0 0 0 0 0 2 1 5 ...
# $ EPM_DistanceTraveled           : num  1428 1851 2574 2471 1807 ...
# $ EPM_Time_Immobile              : num  167 120.7 71.7 81.4 99.4 ...
# $ LearningClassification         : chr  "" "" "" "" ...
# $ PCA_Index_Days6and.7           : num  NA NA NA NA 0.863 ...
# $ Response_Bias_Day7             : num  NA NA NA NA 0.939 ...
# $ Prob_Diff_Day7                 : num  NA NA NA NA 0.92 0.64 -0.8 -0.52 -0.8 -0.12 ...
# $ Latency_Score_Day7             : num  NA NA NA NA 0.614 ...
# $ PCA_Index_Day7only             : num  NA NA NA NA 0.824 ...
# $ Response_Bias_Day6             : num  NA NA NA NA 1 ...
# $ Prob_Diff_Day6                 : num  NA NA NA NA 1 0.32 -0.76 -0.24 -0.8 -0.72 ...
# $ Latency_Score_Day6             : num  NA NA NA NA 0.706 ...
# $ PCA_Index_Day6only             : num  NA NA NA NA 0.902 ...
# $ EPM_RawDataFile                : int  2 2 1 1 2 2 1 1 1 2 ...
# $ EPM_Trial                      : int  16 17 17 18 32 33 4 5 6 13 ...
# $ EPM_TestDate                   : chr  "10/30/2013" "10/30/2013" "10/18/2013" "10/18/2013" ...
# $ EPM_TestDay                    : int  7 7 1 1 7 7 1 1 1 7 ...
# $ EPM_TestOrderByDay             : int  16 17 17 18 32 33 4 5 6 13 ...
# $ Loco_TestDate                  : chr  "9/24/2013" "9/24/2013" "9/26/2013" "9/26/2013" ...
# $ Loco_TestDay                   : int  1 1 3 3 3 3 2 2 2 2 ...
# $ Loco_Batch                     : int  1 1 2 2 2 2 1 1 1 1 ...
# $ Loco_Rack                      : int  1 1 1 1 1 1 1 1 1 1 ...
# $ Loco_Box                       : int  4 5 2 3 4 5 4 5 6 7 ...
# $ DOB                            : chr  "8/6/2013" "8/6/2013" "8/6/2013" "8/6/2013" ...
# $ DOD                            : chr  "12/4/2013" "12/4/2013" "12/13/2013" "12/13/2013" ...
# $ Sex_AsFactor                   : Factor w/ 2 levels "Male","Female": 1 1 2 2 2 2 1 1 1 1 ...
# $ Sex_AsNumeric                  : num  1 1 2 2 2 2 1 1 1 1 ...
# $ Dissector_AsFactor             : Factor w/ 2 levels "EHB","KA": 1 1 1 1 2 2 2 2 2 2 ...
# $ Dissector_AsNumeric            : num  1 1 1 1 2 2 2 2 2 2 ...
# $ HPC_Side_AsFactor              : Factor w/ 2 levels "L","R": 1 2 1 2 2 1 2 1 1 2 ...
# $ HPC_Side_AsNumeric             : num  1 2 1 2 2 1 2 1 1 2 ...
# $ HPC_Dissection_Date_AsFactor   : Factor w/ 10 levels "12/1/2020","12/10/2020",..: 4 4 4 4 1 8 3 2 2 2 ...
# $ LearningClassification_AsFactor: Factor w/ 4 levels "ST","","GT","IN": 2 2 2 2 1 1 3 3 3 4 ...
# $ STGT_experience                : Factor w/ 2 levels "FALSE","TRUE": 1 1 1 1 2 2 2 2 2 2 ...
# $ STGT_experience_AsNumeric      : num  1 1 1 1 2 2 2 2 2 2 ...

library(stringr)

F2_PCA_wMetaData$SeqID_Numeric<-as.numeric(str_replace(F2_PCA_wMetaData$row_name, pattern="sl", replacement=""))
