#F0 HC RNA-Seq Dataset
#01_Reading In MetaData
#Megan Hagenauer and Elaine Hebda-Bauer
#2022-01-10, updated later for a few figures for the paper.


#set working directory:
setwd("//n05-corea-cifs.umhs.med.umich.edu/Home1/Users/hebda/My Documents/F0_TMMnorm_20220110")
getwd()

#Package and version information (updated 1/27/2022):
# sessionInfo()  
# R version 4.1.1 (2021-08-10)
# Platform: i386-w64-mingw32/i386 (32-bit)
# Running under: Windows 10 x64 (build 19042)
# 
# Matrix products: default
# 
# locale:
#   [1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                           LC_TIME=English_United States.1252    
# 
# attached base packages:
#   [1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
#   [1] fgsea_1.18.0         car_3.0-12           carData_3.0-5        plyr_1.8.6           org.Rn.eg.db_3.13.0  edgeR_3.34.1         AnnotationDbi_1.54.1 IRanges_2.26.0       S4Vectors_0.30.2     Biobase_2.52.0       BiocGenerics_0.38.0 
# [12] limma_3.48.3        
# 
# loaded via a namespace (and not attached):
#   [1] Rcpp_1.0.7             locfit_1.5-9.4         lattice_0.20-44        png_0.1-7              Biostrings_2.60.2      assertthat_0.2.1       utf8_1.2.2             R6_2.5.1               GenomeInfoDb_1.28.4    RSQLite_2.2.8         
# [11] httr_1.4.2             ggplot2_3.3.5          pillar_1.6.4           zlibbioc_1.38.0        rlang_0.4.11           rstudioapi_0.13        data.table_1.14.2      blob_1.2.2             Matrix_1.4-0           BiocParallel_1.26.2   
# [21] RCurl_1.98-1.5         bit_4.0.4              munsell_0.5.0          compiler_4.1.1         pkgconfig_2.0.3        tidyselect_1.1.1       KEGGREST_1.32.0        gridExtra_2.3          tibble_3.1.6           GenomeInfoDbData_1.2.6
# [31] fansi_1.0.2            crayon_1.4.2           dplyr_1.0.7            bitops_1.0-7           grid_4.1.1             gtable_0.3.0           lifecycle_1.0.1        DBI_1.1.2              magrittr_2.0.1         scales_1.1.1          
# [41] cachem_1.0.6           XVector_0.32.0         ellipsis_0.3.2         vctrs_0.3.8            generics_0.1.1         fastmatch_1.1-3        tools_4.1.1            bit64_4.0.5            glue_1.6.0             purrr_0.3.4           
# [51] abind_1.4-5            fastmap_1.1.0          colorspace_2.0-2       BiocManager_1.30.16    memoise_2.0.1         


########################################

#Reading in the F0 metadata
#This metadata includes the automated PCA output from Ahub

F0_PCA_wMetaData<-read.delim("HRLR_F0_pca_matrix_RNAconc.txt", sep='\t', header=TRUE, stringsAsFactors = FALSE)

str(F0_PCA_wMetaData)
# 'data.frame':	24 obs. of  23 variables:
#   $ row_name          : chr  "sl483888" "sl483889" "sl483890" "sl483891" ...
# $ Sex               : chr  "Female" "Male" "Male" "Female" ...
# $ Lineage           : chr  "LR" "LR" "HR" "HR" ...
# $ Family            : chr  "L01" "L01" "H02" "H02" ...
# $ Age_days          : int  189 162 163 190 192 165 163 190 165 187 ...
# $ PC1               : num  -0.0118 0.1416 0.2475 0.3507 0.3147 ...
# $ PC2               : num  -0.1134 -0.1208 -0.1083 0.0685 0.0616 ...
# $ PC3               : num  0.1096 0.0784 0.3276 0.2202 -0.0863 ...
# $ PC4               : num  0.0578 0.1502 0.0723 -0.1756 -0.4392 ...
# $ PC5               : num  0.217 -0.1856 -0.3643 0.0563 -0.0302 ...
# $ PC6               : num  -0.076 0.3161 -0.0648 -0.2446 0.0626 ...
# $ Conc              : num  37 51 88.6 81.9 59.8 ...
# $ PF_Reads          : int  24601113 27048765 35810148 31060987 34021183 27895202 27671721 31535526 33899311 35707349 ...
# $ RIN               : num  9.6 8.7 9.3 9.5 9.5 8.8 9.4 9.7 9.6 6.3 ...
# $ DV200             : num  97 96.5 96.9 97.1 97.1 96.5 97.1 96.6 97.2 94.5 ...
# $ RibosomePerc      : num  0.0243 0.0239 0.0199 0.0123 0.0136 ...
# $ mRNAperc          : num  0.829 0.811 0.809 0.85 0.844 ...
# $ Percent_Coding    : num  0.443 0.436 0.429 0.502 0.488 ...
# $ Percent_UTR       : num  0.386 0.375 0.38 0.349 0.356 ...
# $ Percent_Intronic  : num  0.0369 0.0384 0.036 0.0309 0.0334 ...
# $ Percent_Intergenic: num  0.11 0.127 0.135 0.107 0.109 ...
# $ PCT_USABLE_BASES  : num  0.804 0.791 0.793 0.834 0.825 ...
# $ MEDIAN_CV_COVERAGE: num  0.481 0.455 0.447 0.447 0.454 ...

##sl483897 has a low RIN######remove this outlier

F0_PCA_wMetaData<- F0_PCA_wMetaData[(F0_PCA_wMetaData$row_name=="sl483897")==FALSE,]

str(F0_PCA_wMetaData)
#'data.frame':	23 obs. of  23 variables:


#Setting the reference levels for factors:

F0_PCA_wMetaData$Sex_Factor<-as.factor(F0_PCA_wMetaData$Sex)
levels(F0_PCA_wMetaData$Sex_Factor)
#[1] "Female" "Male" 
F0_PCA_wMetaData$Sex_Factor<-relevel(F0_PCA_wMetaData$Sex_Factor, ref="Male")
levels(F0_PCA_wMetaData$Sex_Factor)
#[1] "Male"   "Female"

#bHR and bLR are what we normally use in our publications

F0_PCA_wMetaData$Lineage[F0_PCA_wMetaData$Lineage=="HR"]<-"bHR"
F0_PCA_wMetaData$Lineage[F0_PCA_wMetaData$Lineage=="LR"]<-"bLR"

F0_PCA_wMetaData$Lineage_AsFactor<-as.factor(F0_PCA_wMetaData$Lineage)
levels(F0_PCA_wMetaData$Lineage_AsFactor)
#[1] "bHR" "bLR"


str(F0_PCA_wMetaData)
# 'data.frame':	23 obs. of  25 variables:
#   $ row_name          : chr  "sl483888" "sl483889" "sl483890" "sl483891" ...
# $ Sex               : chr  "Female" "Male" "Male" "Female" ...
# $ Lineage           : chr  "bLR" "bLR" "bHR" "bHR" ...
# $ Family            : chr  "L01" "L01" "H02" "H02" ...
# $ Age_days          : int  189 162 163 190 192 165 163 190 165 189 ...
# $ PC1               : num  -0.0118 0.1416 0.2475 0.3507 0.3147 ...
# $ PC2               : num  -0.1134 -0.1208 -0.1083 0.0685 0.0616 ...
# $ PC3               : num  0.1096 0.0784 0.3276 0.2202 -0.0863 ...
# $ PC4               : num  0.0578 0.1502 0.0723 -0.1756 -0.4392 ...
# $ PC5               : num  0.217 -0.1856 -0.3643 0.0563 -0.0302 ...
# $ PC6               : num  -0.076 0.3161 -0.0648 -0.2446 0.0626 ...
# $ Conc              : num  37 51 88.6 81.9 59.8 ...
# $ PF_Reads          : int  24601113 27048765 35810148 31060987 34021183 27895202 27671721 31535526 33899311 28480670 ...
# $ RIN               : num  9.6 8.7 9.3 9.5 9.5 8.8 9.4 9.7 9.6 8.9 ...
# $ DV200             : num  97 96.5 96.9 97.1 97.1 96.5 97.1 96.6 97.2 96.4 ...
# $ RibosomePerc      : num  0.0243 0.0239 0.0199 0.0123 0.0136 ...
# $ mRNAperc          : num  0.829 0.811 0.809 0.85 0.844 ...
# $ Percent_Coding    : num  0.443 0.436 0.429 0.502 0.488 ...
# $ Percent_UTR       : num  0.386 0.375 0.38 0.349 0.356 ...
# $ Percent_Intronic  : num  0.0369 0.0384 0.036 0.0309 0.0334 ...
# $ Percent_Intergenic: num  0.11 0.127 0.135 0.107 0.109 ...
# $ PCT_USABLE_BASES  : num  0.804 0.791 0.793 0.834 0.825 ...
# $ MEDIAN_CV_COVERAGE: num  0.481 0.455 0.447 0.447 0.454 ...
# $ Sex_Factor        : Factor w/ 2 levels "Male","Female": 2 1 1 2 2 1 1 2 1 2 ...
# $ Lineage_AsFactor  : Factor w/ 2 levels "bHR","bLR": 2 2 1 1 2 2 1 1 1 2 ...


#############################################


#Quickly glancing at the top principal components of variation in the Ahub RNA-Seq data
#Which co-variates seem to be driving them? (or at least strongly related)
#These co-variates are things we should consider including in our differential expression model

F0_PCA_MetaData_CorMatrix<-cor(F0_PCA_wMetaData[,c(5:22)])
write.csv(F0_PCA_MetaData_CorMatrix, "F0_PCA_MetaData_CorMatrix.csv")

#Potential co-variates:

#RNA concentration (strongly related PC1)
#Ribsomal % - also related to PC1, but mostly redundant with RNAconc (so maybe leave out?)
#mRNA % (very strongly related to PC2)
#Sex- might be demanded by reviewers/collaborators/PIs - could be used as a result in itself - results will be in the same output

